src/frontend/js/widgets/Dashboard/components/TeacherCourseSearchFilters/index.tsx

```diff
@@ -0,0 +1,154 @@
+  filterTypePrepend: {
+    defaultMessage: 'Training type:',
+    description: 'Option prepend label for courses search filters',
+    id: 'components.TeacherCourseSearchFiltersBar.type.option.label.prepend',
+  },
+  filterTypeAll: {
+    defaultMessage: 'All',
+    description: 'Option prepend label for courses training types in search filters',
+    id: 'components.TeacherCourseSearchFiltersBar.type.option.label.all',
+  },
+  filterSession: {
+    defaultMessage: 'Session',
+    description: 'Option label for courses session training type in search filters',
+    id: 'components.TeacherCourseSearchFiltersBar.type.option.label.session',
+  },
+  filterMicroCertificate: {
+    defaultMessage: 'Micro certificate',
+    description: 'Option label for courses micro certificate training type in search filters',
+    id: 'components.TeacherCourseSearchFiltersBar.type.option.label.archived',
+  },
+});
+
+export interface TeacherCourseSearchFiltersBarProps {
+  filters: TeacherCourseSearchFilters;
+}
+
+const buildOption = ({
+  label,
+  prependLabel,
+  value,
+}: {
+  label: string;
+  prependLabel: string;
+  value: string;
+}) => (
+  <option value={value} key={value}>
+    {`${prependLabel} ${label}`}
+  </option>
+);
+
+const TeacherCourseSearchFiltersBar = ({ filters }: TeacherCourseSearchFiltersBarProps) => {
+  const intl = useIntl();
+  const navigate = useNavigate();
+  const statusOptions = [
+    {
+      label: intl.formatMessage(messages.filterIncoming),
+      value: CourseStatusFilter.INCOMING,
+    },
+    {
+      label: intl.formatMessage(messages.filterOngoing),
+      value: CourseStatusFilter.ONGOING,
+    },
+    {
+      label: intl.formatMessage(messages.filterArchived),
+      value: CourseStatusFilter.ARCHIVED,
+    },
+  ];
+  const typeOptions = [
+    {
+      label: intl.formatMessage(messages.filterSession),
+      value: CourseTypeFilter.SESSION,
+    },
+    {
+      label: intl.formatMessage(messages.filterMicroCertificate),
+      value: CourseTypeFilter.MIRCO_CERTIFICATE,
+    },
+  ];
+
+  const onChangeFilter = (e: ChangeEvent<HTMLSelectElement>) => {
+    const {
+      target: { name, value },
+    } = e;
+
+    navigate(`?${querystring.stringify({ ...filters, [name]: value })}`, { replace: true });
+  };
+
+  return (
+    <div className="dashboard-course-search-filters">
+      <SelectField
+        fieldClasses={['dashboard-course-search-filters__filter_select']}
+        name="status"
+        id="course_search_filter_status"
+        value={filters.status}
+        onChange={onChangeFilter}
+      >
+        <option value={CourseStatusFilter.ALL}>
+          <FormattedMessage {...messages.filterStatusPrepend} />{' '}
+          <FormattedMessage {...messages.filterStatusAll} />
+        </option>
+        {statusOptions.map(({ label, value }) =>
+          buildOption({
+            label,
+            prependLabel: intl.formatMessage(messages.filterStatusPrepend),
+            value,
+          }),
```

Comment:
It's not really reactish.

```suggestion
<FilterOption key={`status-${label}`} label={label} value={value} prependLabel={...} />
```

Reply:
All our `<SelectField />` directly use the `<option>` tag in a map.
Here i use a function because i feel that's easier to read.
I don't think that we want a components for each option's select.

May be our form elements should use a array of options.

Reply:
I said, we are in JSX context so let's use JSX language so Components here. The output will be the same.